src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/JndiDirectoryDAO.scala

```diff
@@ -54,8 +54,13 @@ class JndiDirectoryDAO(protected val directoryConfig: DirectoryConfig)(implicit
     }
   }
 
-  override def deleteGroup(groupName: WorkbenchGroupName): Future[Unit] = withContext { ctx =>
-    ctx.unbind(groupDn(groupName))
+  override def deleteGroup(groupName: WorkbenchGroupName): Future[Unit] = {
+    listAncestorGroups(groupName).flatMap { ancestors =>
+      if (ancestors.nonEmpty)
```

Comment:
style nit - please always use curly braces, even for 1 line if statements. It makes it easier to read and maintain